src/jdb_to_nwb/convert_video.py

```diff
@@ -1,11 +1,178 @@
+
+    # Calculate velocity and acceleration
+    velocity, acceleration = calculate_velocity_acceleration(position['x'].values, position['y'].values,fps = cam_fps, pixel_to_cm = pixelsPerCm)
+
+    return position['x'].values, position['y'].values, velocity, acceleration
+
+
+def detect_and_replace_jumps(coordinates, pixelJumpCutoff):
+    """
+    Detect and replace jumps in the position data that are bigger than pixelJumpCutoff (default30 cm)
+    Jumps are replaced with NaN
+    """
+    n = len(coordinates)
+    jumps = []
+
+    # Calculate Euclidean distances between consecutive points
+    distances = np.linalg.norm(coordinates[1:] - coordinates[:-1], axis=1)
+    
+    # Find positions where the distance exceeds the threshold: pixelJumpCutoff
+    jump_indices = np.where(distances > pixelJumpCutoff)[0] + 1
+    
+    # Mark all points within the jump range
+    for idx in jump_indices:
+        start = max(0, idx - 1)
+        end = min(n, idx + 2)
+        jumps.extend(range(start, end))
+    
+    # Replace points belonging to jumps with NaN
+    coordinates[jumps] = np.nan
+
+    return coordinates
+
+def fill_missing_gaps(position_data):
+    """
+    Fill missing values in the position data
+    It identifies gaps in the position data and fills them with linear interpolation
+    """
+    # Identify missing values as NaNs
+    missing_values = np.isnan(position_data[:, 0]) | np.isnan(position_data[:, 1])
+
+    # Compute the cumulative sum of missing values to identify contiguous gaps
+    cumulative_sum = np.cumsum(missing_values)
+    gap_starts = np.where(np.diff(cumulative_sum) == 1)[0] + 1
+    gap_ends = np.where(np.diff(cumulative_sum) == -1)[0]
+
+    # Interpolate the missing values in each gap using linear interpolation
+    for gap_start, gap_end in zip(gap_starts, gap_ends):
+        if gap_start == 0 or gap_end == len(position_data) - 1:
+            continue  # ignore gaps at the beginning or end of the data
+        else:
+            x = position_data[gap_start - 1:gap_end + 1, 0]
+            y = position_data[gap_start - 1:gap_end + 1, 1]
+            interp_func = interp1d(x, y, kind='linear')
+            position_data[gap_start:gap_end, 0] = np.linspace(x[0], x[-1], gap_end - gap_start + 1)
+            position_data[gap_start:gap_end, 1] = interp_func(position_data[gap_start:gap_end, 0])
+
+    return position_data
+
+def adjust_video_timestamps(video_timestamps: list, photometry_start_in_arduino_time: float):
+    """Convert video timestamps to corresponding sample number."""
+    # Adjust all arduino timestamps so the photometry starts at time zero
+    video_timestamps = np.subtract(video_timestamps, photometry_start_in_arduino_time)
+
+    return video_timestamps
+
+def calculate_velocity_acceleration(x, y, fps, pixel_to_cm=1):
+    """
+    Calculate velocity and acceleration based on the camera fps and pixelsPerCm
+    """
+    # convert pixel to cm
+    x_cm = x * pixel_to_cm
+    y_cm = y * pixel_to_cm
```

Comment:
I was a bit confused about why this was `pixel_to_cm` - I don't think that should be different from `pixels_per_cm`? I changed it to be same bc thats how we use it. Also removed the default because I don't think default of 1 makes sense?
```suggestion
def calculate_velocity_acceleration(x, y, fps, pixels_per_cm):
    """
    Calculate velocity and acceleration based on the camera fps and pixels_per_cm
    """
    # convert pixel to cm
    x_cm = x * pixels_per_cm
    y_cm = y * pixels_per_cm
```